src/libraries/System.Private.CoreLib/src/System/ReadOnlySpan.cs

```diff
@@ -321,14 +321,25 @@ public bool TryCopyTo(Span<T> destination)
         /// For <see cref="ReadOnlySpan{Char}"/>, returns a new instance of string that represents the characters pointed to by the span.
         /// Otherwise, returns a <see cref="string"/> with the name of the type and the number of elements.
         /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public override string ToString()
         {
             if (typeof(T) == typeof(char))
             {
-                return new string(new ReadOnlySpan<char>(ref Unsafe.As<T, char>(ref _pointer.Value), _length));
+                return string.CreateFromSpan(ref Unsafe.As<T, char>(ref _pointer.Value), _length);
+            }
+            else
+            {
+                return ToStringSlow();
             }
+        }
+
+        private string ToStringSlow()
```

Comment:
I assume this second method is so its aggressively inlined for `char` but not for other Span types?

As its not highly likely to commonly be called for other span types (its output is of low utility); it shouldn't be greatly disadvantageous to have them all as a single method?